File: backend/moyeo/src/main/java/com/mo/moyeo/domain/exchange/rate/entity/ExchangeRate.java
package com.mo.moyeo.domain.exchange.rate.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.mo.moyeo.domain.currency.entity.Currency;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity(name = "exchange_rates")
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
public class ExchangeRate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "rate_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "currency_code")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Currency currency;

    @Column(name = "buy_rate")
    private Double buyRate;

    @Column(name = "sell_rate")
    private Double sellRate;

    @Column(name = "original_rate")
    private Double originalRate;

    @CreationTimestamp
    @Column(name = "recorded_at")
    private LocalDateTime recordedAt;
}

File: frontend/app/src/main/java/com/d108/moyeo/domain/model/exchange/ExchangeRate.kt
package com.d108.moyeo.domain.model.exchange

/**
 * 환율 정보 도메인 모델
 */
data class ExchangeRate(
    val currencyCode: String,       // 통화 코드 (USD, JPY 등)
    val currencyName: String,       // 통화명 (미국 달러, 일본 엔 등)
    val countryFlag: String,        // 국가 플래그
    val buyRate: Double,            // 매수율
    val sellRate: Double,           // 매도율
    val originalRate: Double,       // 기준율
    val changeRate: String,         // 변화율 (+0.59%, -0.32% 등)
    val isIncreased: Boolean        // 상승/하락 여부
)